tensorflow/lite/micro/kernels/leaky_relu.cc

```diff
@@ -92,52 +96,73 @@ TfLiteStatus LeakyReluPrepare(TfLiteContext* context, TfLiteNode* node) {
     TF_LITE_ENSURE_EQ(context, output->params.zero_point, 0);
```

Comment:
its somewhat awkward but there are some implicit float to double conversions happening that break the xtensa build (not part of the CI right now), and we will need some casting to get blupill to pass if we keep the structs as all int32_t.

Here's what I did after checking out this PR:
```cc
    double alpha_multiplier = static_cast<double>(input->params.scale) *
                              static_cast<double>(params->alpha) /
                              static_cast<double>(output->params.scale);

    int output_shift_alpha;
    QuantizeMultiplier(alpha_multiplier, &data->output_multiplier_alpha,
                       &output_shift_alpha);
    data->output_shift_alpha = static_cast<int32_t>(output_shift_alpha);

    double identity_multiplier = static_cast<double>(input->params.scale) /
                                 static_cast<double>(output->params.scale);
    int output_shift_identity;
    QuantizeMultiplier(identity_multiplier, &data->output_multiplier_identity,
                       &output_shift_identity);
    data->output_shift_identity = static_cast<int32_t>(output_shift_identity);
```

The static_cast<double> are only needed with the xtensa toolchain (since it is not reproducible for you, if you miss some of them, that is ok).

The static_cast<int32_t> is for the bluepill target.

I'm going to see if it is possible to get the double promotion errors on xtensa to also be reproducible with either arm-gcc or gcc / clang. If not, I can fix any remaining issues separate from this PR.

Reply:
Fixed